core/src/main/java/com/yahoo/oak/BlockMemoryAllocator.java

```diff
@@ -29,7 +29,7 @@
     long allocated();
 
     // Read the buffer from an allocation object that its parameters are set: blockID, offset and length
-    void readByteBuffer(Slice s);
+    void readByteBuffer(Slice s, int blockID);
```

Comment:
I don't understand the justification for this modification.

Reply:
it is in order to check the new block id with the old

Reply:
As mentioned before, I think this modification is prone to errors.
I suggest moving back updating of the Slice's blockID to the `update()` method, and setting the valid flag after the Slice's buffer was set (not to `null`).